asserts/database.go

```diff
@@ -24,23 +24,52 @@ package asserts
 import (
 	"errors"
 	"fmt"
-	"io/ioutil"
-	"net/url"
 	"os"
 	"path/filepath"
 	"regexp"
 	"strings"
 	"time"
-
-	"github.com/ubuntu-core/snappy/helpers"
 )
 
+// BuilderFromComps can build an assertion from its components.
+type BuilderFromComps func(headers map[string]string, body, content, signature []byte) (Assertion, error)
```

Comment:
Per discussion, we can drop this type and also the Init method in the interface, in favor of a public Assemble function.